File: bto_management_system/model/hdb_officer.py
from .applicant import Applicant # Officers are also Applicants
from common.enums import UserRole

class HDBOfficer(Applicant):
    """Represents an HDB Officer user. Inherits Applicant capabilities."""
    def __init__(self, name: str, nric: str, age: int, marital_status: str, password: str = "password"):
        super().__init__(name, nric, age, marital_status, password)

    def get_role(self) -> UserRole:
        return UserRole.HDB_OFFICER

File: J - Copy (2)/Models/HDBOfficer.java
package Models;

import Enums.UserRole;
import Enums.MaritalStatus;

public class HDBOfficer extends Applicant {

    public HDBOfficer(String nric, String password, String name, int age, MaritalStatus maritalStatus) {
        super(nric, password, name, age, maritalStatus);
    }

    @Override
    public UserRole getRole() { return UserRole.HDB_OFFICER; }

}